File: Backend/DietApp/Domain/Models/Meal.cs
namespace DietApp.Domain.Models
{
    public class Meal
    {
        public int ID { get; set; }
        public string Name { get; set; }
        public Nutrients Nutrients { get; set; }
        public int UserID { get; set; }
        public User User { get; set; }

        public Meal(PublicMeal publicMeal, int userId)
        {
            Name = publicMeal.Name;
            Nutrients = publicMeal.Nutrients;
            UserID = userId;
        }

        public Meal() { }
    }
}

File: Mobile/DietApp/app/src/main/java/com/example/dietapp/models/entity/Meal.kt
package com.example.dietapp.models.entity

import androidx.room.Embedded
import androidx.room.Entity
import androidx.room.PrimaryKey
import com.example.dietapp.models.Nutrients
import com.example.dietapp.models.dto.CreateMealDTO

@Entity(tableName = "meal")
data class Meal(
    @PrimaryKey(autoGenerate = true) val id: Int,
    val name: String,
    @Embedded val nutrients: Nutrients
){
    fun toCreateMealDTO() = CreateMealDTO(name, nutrients)
}